src/components/Add/AddPost.tsx

```diff
@@ -0,0 +1,381 @@
+import React, { useState } from 'react';
+import { z, ZodError } from 'zod';
+import { Button, TextField, Box, Alert, Select, FormControl, InputLabel, MenuItem, Chip } from '@mui/material';
+import SendIcon from '@mui/icons-material/Send';
+import { Post } from '../../types/post';
+import { randomId } from '../../utils/createId';
+import { useDispatch } from 'react-redux';
+import { postSlice } from '../../app/redux/features/posts/post-slice';
+import ReactQuill from 'react-quill';
+import 'react-quill/dist/quill.snow.css';
+import './AddPost.scss';
+import { CloudUpload } from '@mui/icons-material';
+
+const schema = z.object({
+  title: z
+    .string()
+    .min(4, {
+      message: 'Title needs to be at least 4 characters',
+    })
+    .max(30)
+    .regex(/^[\w,.?! -]+$/, {
+      message:
+        'Title can only contain letters, commas, dots, question marks, exclamation marks, hyphens, and spaces',
+    }),
+  content: z
+    .string()
+    .refine((value) => value.length >= 40, {
+      message: 'Content must be at least 40 characters',
+    }),
+  tags: z.array(z.string()).refine((value) => value.length <= 6, {
+    message: 'You can\'t add more than 6 tags',
+  }),
+  category: z.string().refine((value) => value.trim() !== '', {
+    message: 'Please select a category.',
+  }),
+});
+
+const AddPost: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState<string | null>(null);
+  const [content, setContent] = useState('');
+  const [contentError, setContentError] = useState<string | null>(null);
+  const [tags, setTags] = useState<string[]>([]);
+  const [tagInput, setTagInput] = useState('');
+  const [tagsError, setTagsError] = useState<string | null>(null);
+  const [uploadedImage, setUploadedImage] = useState<File | null>(null);
+  const [imageError, setImageError] = useState<string | null>(null);
+  const [category, setCategory] = useState<string>('');
+  const [categoryError, setCategoryError] = useState<string | null>(null);
+  const dispatch = useDispatch();
+
+  const onSubmit = async (data: Post) => {
+    try {
+      const response = await fetch('http://localhost:3000/posts/addPost', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+  
+      if (response.ok) {
+        const responseData = await response.json();
+        dispatch(postSlice.actions.addPost(responseData));
+        // Czyszczenie stanów po pomyślnym dodaniu postu
+        setTitle('');
+        setContent('');
+        setTags([]);
+        setUploadedImage(null);
+        setCategory('');
+        // Ustawienie błędów na null
+        setTitleError(null);
+        setContentError(null);
+        setTagsError(null);
+        setImageError(null);
+        setCategoryError(null);
```

Comment:
Dobra po tym i po fakcie ile masz useState wnosze o to zeby to rozbic jako objekt z jakas defaultową strukturą zmniejszając tym liczbę useState.